app/src/main/scala/com/waz/zclient/appentry/SSOFragment.scala

```
@@ -42,20 +45,21 @@ trait SSOFragment extends FragmentHelper with DerivedLogTag {
   private lazy val ssoService             = inject[SSOService]
   private lazy val userAccountsController = inject[UserAccountsController]
 
-  private lazy val dialogStaff = new InputDialog.Listener with InputDialog.InputValidator {
+  private lazy val inputValidator = EnterpriseLoginInputValidator(ssoService)
+
+  private lazy val dialogStaff = new InputDialog.Listener {
     override def onDialogEvent(event: Event): Unit = event match {
-      case OnNegativeBtn        => verbose(l"Negative")
-      case OnPositiveBtn(input) => verifyInput(input)
+      case OnPositiveBtn(input) => inputValidator.inputType match {
+        case Some(SsoCode) => verifySsoCode(input)
+        case Some(Email) => verifyEmail(input)
+        case _ => throw new IllegalStateException("User should not be able to click the button without a valid input")
```

Can we reorganize the code so that it won't be possible?

---

actually, there is no way that this case is reached. because if the input is not an sso code nor an email, the positive button won't get enabled and this callback wouldn't be invoked. I just added this to make the match exhaustive and to fail fast if sth changes, rather than staying unresponsively